### PR TITLE
adding a network testing case

### PR DESCRIPTION
**Pull request by GitPaean**
it is based on model 2 while with some additions and modifications.

it was created by @vkip originally, while we modified it later along development.

### CODE REVIEW

File: model2/NETBALAN_P80.DATA [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+
+-- Copyright (C) 2022 Equinor
+
+
+-- ------------------------------------------------------------------------
+-- ----------------------- NETWORK TESTING -----------------------------------
+-- ------------------------------------------------------------------------
+
```
Comment (bska) [inline model2/NETBALAN_P80.DATA:12]: Could you add a couple of lines here to explain the features we're testing in this model?  Just "network testing" is a little too broad in my opinion.

**Review by bska — COMMENTED** [review]
This looks good in isolation if we just add a little more description of the feature/situation under test to the `.DATA` file.  That said, the `model2` directory is getting a little crowded by now so I think we should consider adding this to a new directory, e.g., `network`, instead.  Doing so will require duplicating some files from `model2/include`, but that's a price I think I'm willing to pay.

**Comment by GitPaean** [comment]
> Doing so will require duplicating some files from model2/include, but that's a price I think I'm willing to pay.

Thanks for the suggestion. We will do that.

**Comment by OPMUSER** [comment]
@GitPaean I would suggest that you put the test it in the `network`  directory and call the case `network-01`, that way it is easy to add additional tests and compare the results against E100. 

Although I disagree, the consensus is that we do not use the `PATH` keyword to load in the base models. So that means you have to copy all model's include file into the `network/include` directory. Also rename the include files to `MODEL01-PERMX.inc`, etc so one knows which model is being used. Take a look at the `grupcnt` directory for an example - feel free to use these file if useful.

**Comment by GitPaean** [comment]
@bska @OPMUSER  can you help to have a look at the updated version of the PR? Maybe I am not good at it, formatting files can take a lot of efforts and can be error pruning. I need to borrow more eyes here.

**Comment by OPMUSER** [comment]
This looks better, can you rename [flowline_netbalance.inc](https://github.com/GitPaean/opm-tests/blob/adding_network_test/network/include/flowline_netbalance.inc) to` NETWORK-01-VFPPROD.inc `so that we know which DATA file  is using the file. Don't forget to change the DATA file as well.

The DATA file itself is a bit messy, but if @bska comments are addressed then  it should be merged as is, and I'll clean it up when I get a minute. Easy and quicker for me to do.

**Comment by GitPaean** [comment]
@bska @OPMUSER , can you let me know whether we can get this in or what more I need to do? 

Thanks.

**Comment by bska** [comment]
> can you let me know whether we can get this in or what more I need to do?

I'm actually happy with the current state, and I think `flowline_netbalance` is a better filename than `NETWORK-01-VFPPROD` since the latter highlights the keyword rather than the purpose.  I'll merge into master.

**Comment by OPMUSER** [comment]
> I'm actually happy with the current state, and I think flowline_netbalance is a better filename than NETWORK-01-VFPPROD since the latter highlights the keyword rather than the purpose. I'll merge into master.

I suggested the change because if and when we add additional tests to this directory, it will be easier to track what files go with a particular DATA file. But we can fight that battle another day :-)

**Pull request merged**